src/core/queriesObserver.ts

```diff
@@ -64,57 +64,94 @@ export class QueriesObserver extends Subscribable<QueriesObserverListener> {
   }
 
   getOptimisticResult(queries: QueryObserverOptions[]): QueryObserverResult[] {
-    return queries.map((options, index) => {
-      const defaultedOptions = this.client.defaultQueryObserverOptions(options)
-      return this.getObserver(defaultedOptions, index).getOptimisticResult(
-        defaultedOptions
-      )
-    })
+    return this.findMatchingObservers(queries).map(match =>
+      match.observer.getCurrentResult()
+    )
```

Comment:
quick follow-up question here @jvuoti : previously, we called `getOptimisticResult` of each observer, and now we just call `getCurrentResult()`. I'm surprised that this doesn't seem to make a difference 🤔 ...

Reply:
Ah, you are right, it should call `getOptimisticResult` there, so that is a bug. Sorry, must have been some lazy copy-pasting on my part when refactoring the code around. 🤦 

Testing this out quickly, the change does not seem to cause any visible effects - but it _does_ now fix the issue I had with some of the existing tests still returning `isFetching: false`. So yeah, I probably should have noticed the issue from those tests. But at least now I can revert the tests back to the original, correct form 😅 

So good catch, thank you so much! I'll prep a PR to fix the call and tests shortly.

Reply:
Thank you 🙏